routes/auth.routes.js

```diff
@@ -0,0 +1,119 @@
+const router = require("express").Router();
+const bcrypt = require("bcryptjs");
+const User = require("../models/User.model");
+const mongoose = require("mongoose");
+const { isLoggedIn, isLoggedOut } = require('../middleware/route-guard.js');
+
+router.get("/signup", isLoggedOut, (req, res, next) => {
+  res.render("auth/signup")
+});
+
+router.post("/signup", isLoggedOut, (req, res, next) => {
+  const { username, password } = req.body;
+
+
+  if (!username || !password) {
+    res.render("auth/signup", {
+      errorMessage: "All fields are mandatory. Please provide your username and password."
+    });
+    return;
+  }
+
+  // code imported from the student portal, need to clarify this step
+  const regex = /(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,}/;
+  if (!regex.test(password)) {
+    res
+      .status(500)
+      .render('auth/signup', { errorMessage: 'Password needs to have at least 6 chars and must contain at least one number, one lowercase and one uppercase letter.' });
+    return;
+  }
```

Comment:
Regex will test if the password has certain requirements, in this case will be 6 chars, 1 number, one lowercase and one uppercase letter